### PR TITLE
Convert to spies

### LINKED ISSUE #9: Code Cleanup: Use busted spies instead of "callCount"
Basically everything related to callbacks in the code uses a variable to track function calls. They litter the signal spec code.

https://github.com/LPGhatguy/lemur/blob/705238b707d48ee8ce926814aa0cf22393caa32e/lib/Signal_spec.lua

Busted provides something called "spies" for this exact usage, where it tracks if a given function is called. This would help make the code more readable.

### PR DESCRIPTION
**Pull request by AmaranthineCodices**
Fixes #9. Not much else to say.

### CODE REVIEW

File: lib/Signal_spec.lua [hunk 1]
```diff
@@ -9,52 +9,41 @@ describe("Signal", function()
...
-			callCount = callCount + 1
-		end)
+		local connection = signal:Connect(spy)
 
 		assert.not_nil(connection)
-		assert.equal(callCount, 0)
 
 		signal:Fire()
-		assert.equal(callCount, 1)
+		assert.spy(spy).was_called()
```
Comment (LPGhatguy) [inline lib/Signal_spec.lua]: What's the default behavior for `was_called`? To me, this reads "this function was called *at least once*", but the behavior we want is "this function was called *exactly once*"

Does this need to change to `was_called(1)`?

Comment (Kampfkarren) [reply lib/Signal_spec.lua]: https://github.com/Olivine-Labs/luassert/blob/73b118e86ca40f1db6dc92b300a68d657c36b39b/src/spy.lua#L51

Yes, it's called at least once. This needs to be changed.

File: lib/Signal_spec.lua [hunk 2]
```diff
@@ -9,52 +9,41 @@ describe("Signal", function()
...
-		assert.equal(countB, 0)
+		local connectionA = signal:Connect(spyA)
+		signal:Connect(spyB)
 
 		signal:Fire()
 
-		assert.equal(countA, 1)
-		assert.equal(countB, 1)
+		assert.spy(spyA).was_called()
+		assert.spy(spyB).was_called()
```
Comment (Kampfkarren) [inline lib/Signal_spec.lua]: Change these also to was_called(1) (and any other ones there are)

File: lib/Signal_spec.lua (hunk 1 above)
Comment (AmaranthineCodices) [reply lib/Signal_spec.lua]: Fixed now!

**Comment by LPGhatguy** [comment]
Awesome! Thanks for the fix!

**Pull request merged**